src/crypto-api.ts

```diff
@@ -599,7 +613,8 @@ export interface ImportRoomKeysOpts {
     progressCallback?: (stage: ImportRoomKeyProgressData) => void;
     // TODO, the rust SDK will always such imported keys as untrusted
     untrusted?: boolean;
-    source?: String; // TODO: Enum (backup, file, ??)
+    /** Where the room key came from */
+    source?: RoomKeySource;
```

Comment:
Hrm. I don't think this should be part of the public API to the JS-SDK: the only sensible value for an external caller is `Export`.

Unfortunately, removing `source` is technically an API-breaking change, (as is changing its type, fwiw,) and we try to avoid them.

What we *can* do is something like:

* Mark `source` as `@deprecated`. Might as well do `untrusted` while we're here.
* Add a new *internal* method to `CryptoBackend` called `importRoomKeysFromBackup` or somesuch. `CryptoApi.importRoomKeys` is only for importing from export.
* Change existing codepaths in the js-sdk which use `importRoomKeys` to import from backup to use the new method.
* In the rust impl, the two methods can be quite separate. In the legacy impl, you probably want to make one call the other. Hack it together in whatever way is easiest.

WDYT?

Reply:
Yup, I considered adding a separate function, but opted to make fewer changes to the API.  But I can do so, and get rid of an `if` statement.

Reply:
we can just throw an error if the `source` is set to something else.